code/control/DNRoot.php

```diff
@@ -2339,4 +2346,99 @@ public function AmbientMenu() {
 		return $list;
 	}
 
+	/**
+	 * Create project action.
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function createproject(SS_HTTPRequest $request) {
+		if($this->canCreateProjects()) {
+			return $this->render(['CurrentTitle' => 'Create Stack']);
+		}
+		return $this->httpError(403);
+	}
+
+	/**
+	 * Checks whether the user can create a project.
+	 *
+	 * @return bool
+	 */
+	protected function canCreateProjects() {
+		$project = Injector::inst()->get('DNProject');
+		return $project->allowed(self::ALLOW_CREATE_PROJECT);
+	}
+		
+	/**
+	 * @return Form
+	 */
+	public function CreateProjectForm() {
+		$form = Form::create(
+			$this, 
+			__FUNCTION__, 
+			$this->getCreateProjectFormFields(), 
+			$this->getCreateProjectFormActions(),
+			new RequiredFields('Name', 'CVSPath')
+		);
+		$this->extend('updateCreateProjectForm', $form);
+		return $form;
+	}
+
+	/**
+	 * @return FieldList
+	 */
+	protected function getCreateProjectFormFields() {
+		$fields = FieldList::create();
+		$fields->merge([
+			TextField::create('Name', 'Title'),
+			TextField::create('CVSPath', 'Git URL'),
+			TextField::create('Client', 'Client'),
+		]);
+		$this->extend('updateCreateProjectFormFields', $fields);
+		return $fields;
+	}
+
+	/**
+	 * @return FieldList
+	 */
+	protected function getCreateProjectFormActions() {
+		$fields = FieldList::create(
+			FormAction::create('doCreateProject', 'Create Project')
+		);
+		$this->extend('updateCreateProjectFormActions', $fields);
+		return $fields;
+	}
+
+	/**
+	 * Does the actual project creation.
+	 *
+	 * @param $data array
+	 * @param $form Form
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function doCreateProject($data, $form) {
+		$form->loadDataFrom($data);
+		$project = DNProject::create();
+
+		$form->saveInto($project);
+		$this->extend('onBeforeCreateProject', $project, $data, $form);
+		try {
+			if($project->write() > 0) {
```

Comment:
Do we log any metrics in this module? Seems a strange place to be logging metrics.